src/analysis/normed_space/pointwise.lean

```diff
@@ -0,0 +1,177 @@
+/-
+Copyright (c) 2021 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+
+import analysis.normed_space.basic
+import topology.metric_space.hausdorff_distance
+
+/-!
+# Properties of pointwise addition and scalar multiplication of sets in normed spaces.
+
+We explore the relationships between pointwise addition and scalar multiplication of sets in
+vector spaces, and the norm. Notably, we express arbitrary balls as rescaling of other balls,
+and we show that the sum or multiplication of bounded sets remain bounded.
+-/
+
+open metric set
+open_locale pointwise topological_space
+
+section semi_normed_group
+
+variables {E : Type*} [semi_normed_group E]
+
+lemma metric.bounded.add
+  {s t : set E} (hs : bounded s) (ht : bounded t) :
+  bounded (s + t) :=
+begin
+  obtain ⟨Rs, hRs⟩ : ∃ (R : ℝ), s ⊆ closed_ball 0 R := hs.subset_ball 0,
+  obtain ⟨Rt, hRt⟩ : ∃ (R : ℝ), t ⊆ closed_ball 0 R := ht.subset_ball 0,
+  refine (bounded_iff_subset_ball 0).2 ⟨Rs + Rt, _⟩,
+  rintros z ⟨x, y, hx, hy, rfl⟩,
+  rw mem_closed_ball_zero_iff,
+  calc ∥x + y∥ ≤ ∥x∥ + ∥y∥ : norm_add_le _ _
+  ... ≤ Rs + Rt :
+    add_le_add (mem_closed_ball_zero_iff.1 (hRs hx)) (mem_closed_ball_zero_iff.1 (hRt hy))
+end
+
+@[simp] lemma singleton_add_ball (x y : E) (r : ℝ) :
+  {x} + ball y r = ball (x + y) r :=
+by simp only [preimage_add_ball, image_add_left, singleton_add, sub_neg_eq_add, add_comm y x]
```

Comment:
I think it's worth having
```lean
@[simp] lemma ball_add_singleton (x y : E) (r : ℝ) :
  ball x r + {y} = ball (x + y) r := sorry
```
just so that `simp` doesn't only know how to deal with one associativity.